tethysext/atcore/controllers/resource_workflows/map_workflows/spatial_input_mwv.py

```diff
@@ -458,7 +458,34 @@ def post_process_geojson(geojson):
             return geojson
 
         # Sort the features for consistent ID'ing
-        s_features = sorted(geojson['features'], key=lambda f: f['geometry']['coordinates'])
+        min_sort_vals = []
+        for idx, feature in enumerate(geojson['features']):
+            # Get the min coordinates for each feature, depending on feature type
+            cur_x = []
+            cur_y = []
+            geometry = feature['geometry']
+            for coordinates in geometry['coordinates']:
+                if type(coordinates) == float:
+                    # Point feature
+                    cur_x.append(geometry['coordinates'][0])
+                    cur_y.append(geometry['coordinates'][1])
+                elif type(coordinates) == list:
+                    for val in coordinates:
+                        if type(val) == float:
+                            # LineString feature
+                            cur_x.append(coordinates[0])
+                            cur_y.append(coordinates[1])
+                        elif type(val) == list:
+                            # Polygon and Extent feature
+                            cur_x.append(val[0])
+                            cur_y.append(val[1])
+            cur_val = (min(cur_x), min(cur_y), idx) if cur_x and cur_y else (None, None, idx)
+            min_sort_vals.append(cur_val)
+        # Sort the min coordinates, and then store the features in that order
+        s_features = []
+        sorted_values = sorted(min_sort_vals)
+        for sorted_value in sorted_values:
+            s_features.append(geojson['features'][sorted_value[2]])
```

Comment:
Instead of looping and sorting manually, put the type-checking logic in a function and pass that function to sorted. In Python, functions can be defined in nested scopes like this:

```python
def sort_by_coordinates(f):
    coordinates = f['geometry']['coordinates']
    geom_type = f['geometry']['type']  # Pretty sure there is a type attribute
    if geom_type == 'POLYGON':
        return # get the first coordinate pair
    elif geom_type == 'POLYLINE':
        return # return the first coordinate pair
    elif geom_type == 'POINT':
        return coordinates

s_features = sorted(geojson['features'], key=sort_by_coordinates)
```